qt/scientific_interfaces/Direct/ALFInstrumentPresenter.cpp

```diff
@@ -94,14 +138,52 @@ void ALFInstrumentPresenter::notifyTubesSelected(std::vector<DetectorTube> const
   }
 }
 
+std::optional<std::string> ALFInstrumentPresenter::getFileFromView() const {
+  switch (m_dataSwitch) {
+  case ALFData::SAMPLE:
+    return m_view->getSampleFile();
+  case ALFData::VANADIUM:
+    return m_view->getVanadiumFile();
+  }
+  throw std::invalid_argument("ALFData must be one of { SAMPLE, VANADIUM }");
```

Comment:
I was going to suggest adding a test to check these invalid argument errors get thrown, but I think it's currently unreachable since both values of `m_dataSwitch`. Still good to keep in case someone changes `ALFData`.